src/matrix/kaldi-matrix.h

```diff
@@ -324,6 +324,14 @@ class MatrixBase {
   /// the same).
   void AddToRows(Real alpha, Real *const *dst) const;
 
+  /// For each row i of *this, adds this->Row(i) to
+  /// dst->Row(indexes(i)) if indexes(i) >= 0, else do nothing.
+  /// Requires that all the indexes[i] that are >= 0
+  /// be distinct, otherwise the behavior is undefined.
+  void AddToRows(Real alpha,
```

Comment:
In general I don't like adding too many of this type of function in class Matrix, because many of these functions only make sense in the GPU context (on CPU you could just make a loop).  But because the other AddToRows() function already existed in class Matrix, it probably makes sense to put this one there too, for symmetry.